src/library/scala/util/Random.scala

```diff
@@ -70,6 +70,16 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextLong(): Long = self.nextLong()
 
+  def nextLong(n: Long): Long = {
+    if (n <= 0) throw new java.lang.IllegalArgumentException("bound must be positive")
+    if (n <= Int.MaxValue) nextInt(n.toInt).toLong
+    else {
+      val RAND_MAX = Long.MaxValue     // 2^63-1
+      val x = nextLong() >>> 1         // [0, 2^63), strip the sign bit
+      if (x < (RAND_MAX - (RAND_MAX % n))) x % n else nextLong(n)
```

Comment:
Interesting about the div/mod - thank you for the explanation.

I'm wondering whether it's worth even pursuing this PR, given your statement about this not working well with scala.js, which is, I guess, a soft requirement for any scala code. The main issue that I have with the scala.js code is that it doesn't look like any code I've seen that is designed to provide a uniform random distribution. That's not to say it *doesn't*, but the code I've proposed is based off / inspired by Schneier and other sources I trust. For me, it's more important to have a uniform distribution with this function than to eke out every single optimization. Perhaps that's not the most important thing for others, which is why I'm now questioning the appropriateness of the PR.

Let me know - I'm happy to close this out and just use the code locally.